### PR TITLE
Configure i18n for Portaljs

### PR DESCRIPTION
**Pull request by steveoni**
This pull request configures and setup i8n for Portal js. The configuration leads to the upgrading of nextjs from version 9.x to 10, due to the fact that it is very difficult to set up i18n for server-side Props.

The latest version of Nextjs gives the ability to switch from a different language using subpath, this makes providing language namespace easier.

## Usecase
For subsequent users to add their language of choice, below are the steps to take

1.  Update ` next.config.js`, to add more languages to the i18n locales
```js
i18n: {
  locales: ['en', 'fr', 'nl-NL'], // add more language to the list
  defaultLocale: 'en',  // set the default language to use
},
```
2. Create a folder for the language in `locales`  --> `locales/en-Us`

3. In the language folder, different namespace files (json) can be created for each translation. For the `index.js` use-case, I named it `common.json`
```json
// locales/en/common.json
{
   "title" : "Portal js in English",
}

// locales/fr/common.json
{
   "title" : "Portal js in French",
}
```

4. To used on pages using Server-side Props.
```js
import { loadNamespaces } from './_app';
import useTranslation from 'next-translate/useTranslation';

const Home: React.FC = ()=> {
  const { t } = useTranslation();
  return (
    <div>{t`common:title`}</div> // we use common and title base on the common.json data
  );
};

export const getServerSideProps: GetServerSideProps = async ({ locale }) => {
      ........  ........
  return {
    props : {
      _ns:  await loadNamespaces(['common'], locale),
    }
  };
};

```
5. Go to the browser and view the changes using language subpath like this. `http://localhost:3000` and `http://localhost:3000/fr`. **Note** The subpath also activate chrome language Translator

### CODE REVIEW

**Comment by rufuspollock** [comment]
@steveoni looks good.

* can you rebase so we can merge (merge conflict atm)
* can you add some info to the README about how to use this (similar to what you have in PR description)

**Comment by steveoni** [comment]
> @steveoni looks good.
> 
> * can you rebase so we can merge (merge conflict atm)
> * can you add some info to the README about how to use this (similar to what you have in PR description)

ok. I will do that now.

**Comment by steveoni** [comment]
The conflict is now resolved and the readme is updated @rufuspollock

File: packages/portal/README.md [hunk 1]
```diff
@@ -199,6 +199,61 @@ export default function Org({ variables }) {
 }
 ```
 
+#### i18n configuration
+
+Portaljs is configured by default to support both `English` and `french` subpath for language translation. But for subsequent users, this following steps can be use to configure i18n for other languages;
```
Comment (anuveyatsu) [inline packages/portal/README.md:204]: @steveoni could you fix spelling of:

- "Portal.js" so we have `.`
- "French" so it's capitalized
- "can be used" so it is passive voice

File: packages/portal/README.md [hunk 2]
```diff
@@ -199,6 +199,61 @@ export default function Org({ variables }) {
 }
 ```
 
+#### i18n configuration
```
Comment (anuveyatsu) [inline packages/portal/README.md:202]: @steveoni I think this section should be sibling of "Data fetching" section and should come afterwards.

File: packages/portal/README.md [hunk 3]
```diff
@@ -199,6 +199,61 @@ export default function Org({ variables }) {
...
+   "title" : "Portal js in English",
+}
+
+// locales/fr/common.json
+{
+   "title" : "Portal js in French",
+}
+```
+
+4. To used on pages using Server-side Props.
```
Comment (anuveyatsu) [inline packages/portal/README.md:231]: "To used" => "To use"

File: packages/portal/README.md [hunk 4]
```diff
@@ -199,6 +199,61 @@ export default function Org({ variables }) {
...
+  return {
+    props : {
+      _ns:  await loadNamespaces(['common'], locale),
+    }
+  };
+};
+
+```
+
+5. Go to the browser and view the changes using language subpath like this. `http://localhost:3000` and `http://localhost:3000/fr`. **Note** The subpath also activate chrome language Translator
```
Comment (anuveyatsu) [inline packages/portal/README.md:255]: Colon instead of period before URLs

**Comment by anuveyatsu** [comment]
@steveoni I have left some comments. Also, please remove `package-lock.json` file as we are using `yarn.lock` and `yarn` by default. Having both lock files is risky as they can easily become out of sync.

**Comment by anuveyatsu** [comment]
@steveoni any updates on this PR? Please, check my review.

**Comment by steveoni** [comment]
> @steveoni any updates on this PR? Please, check my review.

I thought I'd made the changes last week but I've not. I'm sorry for that. I will do that today

**Comment by steveoni** [comment]
@anuveyatsu

**Review by anuveyatsu — APPROVED** [review]
@steveoni thanks

**Pull request merged**